.travis.yml

```diff
@@ -38,13 +38,6 @@ before_install:
           jython -c "print ''";
           jython -c "import sys; print sys.version"
         fi
-        if [[ $TRAVIS_OS_NAME == 'osx' ]]; then
-          brew update
-          brew upgrade
-          brew upgrade python
-          brew install python3
-          python3 --version
-        fi
```

Comment:
I saw your experimental branch with the newer macOS image. I came to the same conclusion about the default being easier, although travis isn't doing a great job either way. It would also be good to test on Python 3.7.

I think that should happen in a separate PR, though, so this one can remain focussed on cleaning up warnings in the test output.